packages/terra-switch/src/Switch.jsx

```diff
@@ -0,0 +1,145 @@
+import React, { useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+import uniqueid from 'lodash.uniqueid';
+import classNames from 'classnames/bind';
+import * as KeyCode from 'keycode-js';
+import styles from './Switch.module.scss';
+
+const cx = classNames.bind(styles);
+
+const SWITCH_STATE = Object.freeze({
+  ON: 'On',
+  OFF: 'Off',
+});
+
+const propTypes = {
+  /**
+    * Whether or not the Switch is enabled ("ON").
+    */
+  isOn: PropTypes.bool,
```

Comment:
Is there a conversation or discussion that lead to choosing this prop name? A brief look around similar component libraries the vast majority use `checked`. We don't necessarily need to conform to using `checked`, but we should have a strong argument for using `isOn` if this is the prop name we decide on.

https://ant.design/components/switch/
https://material-ui.com/components/switches/
http://react-toolbox.io/#/components/switch
https://evergreen.segment.com/components/switch

Reply:
I have tried to align this prop name with  switch state ( on and off ) and I feel `checked` seems to be more appropriate for checkbox. there was no discussion regarding prop-name `isOn` on tech-design, but we can change it to `checked` or whichever is more precise for component.